lib/axiom/fs/base/file_system.js

```diff
@@ -205,11 +205,16 @@ FileSystem.prototype.readFile = function(
   var whence = opt_whence || SeekWhence.Begin;
 
   return this.createOpenContext(path, openMode).then(
-      function(cx) {
-        return cx.open().then(function() {
-          return cx.read(offset, whence, dataType);
-        });
-  });
+    function(cx) {
+      return OpenContext.scope(cx, function() {
+        return cx.open().then(
+          function() {
```

Comment:
Gotcha.  Yeah.  I'm seeing it all over, now.  I guess we need to clarify this aspect (not for this, but for future cases).  I'm seeing both styles throughout our code.

Cc @gaurave @ussuri @rginda